packages/lib/models/settings/settingValidations.ts

```diff
@@ -37,24 +38,47 @@ const validations: Record<string, ValidationHandler> = {
 
 		return '';
 	},
+};
+
+const validateInteger = async (key: string, newValue: string) => {
+	const md = Setting.settingMetadata(key);
+	const minimum = 'minimum' in md ? md.minimum : 0;
+	const maximum = 'maximum' in md ? md.maximum : 10;
+	const newValueInt = Math.floor(Number(newValue));
```

Comment:
This comment it pivotal for most of your other comments, so I'll just answer this one.

The reason for formatting on validation is because in order to properly validate the input, you need to first store the input as a string. If for example you enter 1.2, if you used the integer type, then it will truncate the decimal in Setting.formatValue and store it as 1, rather than failing. So because we set the value on change, but validate on save, you kind of have to use a different type for set and validate.

I could set the value as an integer in both set and validate if that's what you would like. But bear in mind that is going to reduce the quality of the validation. As it will truncate a decimal instead of raising a validation error. As per your comment on the error message, the reason I used the word 'integer' instead of 'number', is because entering the value 1.2, then getting the error '1.2 must be a valid number...' is going to be confusing for the user, because that would be a valid number.

Reply:
Ok, I've managed to find a workaround for this. If I check if the value is an integer before updating the setting, I can force it to a null value in order to make it fail the validation